R/stochastic_process.R

```diff
@@ -407,22 +408,22 @@ read_xz_csv <- function(con, the_file, outcomes, allow_missing_disease,
 
   csv$country <- countries$nid[match(csv$country, countries$id)]
 
-  if (is.data.frame(outcomes$dalys)) {
+  if (is.data.frame(dalys_recipe)) {
     res <- stoner_calculate_dalys(con, touchstone, csv,
-                                  outcomes$dalys, cache$life_table)
+                                  dalys_recipe, cache$life_table)
     csv <- res$data
     if (is.null(cache$life_table)) {
       cache$life_table <- res$life_table
     }
-
-  } else {
-    csv <- calc_outcomes(csv, outcomes$dalys, "dalys")
   }
 
-  csv <- calc_outcomes(csv, outcomes$deaths, "deaths")
-  csv <- calc_outcomes(csv, outcomes$cases, "cases")
+  for (i in seq_along(outcomes)) {
+    csv <- calc_outcomes(csv, outcomes[[i]], names(outcomes)[i])
+  }
 
-  csv[, c("run_id", "year", "age", "country", "deaths", "cases" ,"dalys")]
+  csv <- as.data.frame(csv)
+  csv[, unique(c("run_id", "year", "age", "country", names(outcomes),
+          if (!is.null(dalys_recipe)) "dalys"))]
```

Comment:
For readability, pull this into an assignment and index:

```suggestion
  cols <- unique(c("run_id", "year", "age", "country", names(outcomes), if (!is.null(dalys_recipe)) "dalys"))
  csv[, cols]
```

but if csv is a data.frame please use `csv[cols]`